### ISSUE TITLE
Remove the LogLikelihood class

### ISSUE THREAD

**Issue by MichaelClerx** [pints-team/pints#457]
It doesn't add any functionality, just indicated that a certain LogPDF is intended as a LogLikelihood.

In some cases, we test to see if something _is_ a LogLikelihood, but I'm wondering if those checks shouldn't be replaced by a check to see if it's just any old LogPDF...

**Comment by martinjrobins** [comment]
I can't see any reason to keep LogLikelihood, its only in there since we didn't have a LogPDF at the time I think.

**Comment by mirams** [comment]
I would be expecting a pdf to integrate to one, but not a likelihood...

**Comment by MichaelClerx** [comment]
https://pints.readthedocs.io/en/latest/log_pdfs.html

Nothing needs to sum to 1 in Pints! We do say that a PDF probably doesn't, and a Prior probably does, but that's about it.

**Comment by MichaelClerx** [comment]
More importantly, why were you two discussing this on a sunday night before a bank holiday :p

**Comment by ben18785** [comment]
I like keeping it because it is part of the (classical and Bayesian) inference process.

**Comment by MichaelClerx** [comment]
I'm still on the fence! Notice how I left it out in yesterday's diagram and nobody complained...

**Comment by sanmitraghosh** [comment]
@MichaelClerx  I think the the functionality is important, thus externally a user can access the likelihood explicitly as well as internally (I have done this in the power_posterior branch where I have raised the likelihood to the power) inside an algorithm such as smc. You can get rid of a separate class as long as this functionality is intact.

**Comment by mirams** [comment]
Can we just rename `logPDF` to `logLikelihood` if the thing really is a non-normalised likelihood and not a PDF? Or would it be too confusing to consider priors and posteriors to inherit from (be special cases of) a likelihood?

**Comment by MichaelClerx** [comment]
Right. Functionally it's just a label right now. And the LogPDF documentation clearly states its not necessarily normalised (how could it be, if it's the parent class of LogLikelihood). So I'm going to remove it!

**Comment by MichaelClerx** [comment]
Re-opening as I think the distinction matters for transformations.

https://theoryandpractice.org/stats-ds-book/distributions/invariance-of-likelihood-to-reparameterizaton.html

If I understand it correctly (big if), then a likelihood is a function of a non-random variable, while a general unnormalised log pdf is a function of a random variable. This means that when you apply a transformation to the parameter space, you need to take a jacobian into account for general pdfs, but not for likelihoods.
